packages/storybook/src/stories/4-ContractEditor.stories.js

```diff
@@ -45,22 +47,42 @@ This is text. This is *italic* text. This is **bold** text. This is a [link](htt
     return slateTransformer.fromMarkdown(markdownText).document.children;
   });
 
-  useEffect( () => {
+  useEffect(async () => {
       Template.fromUrl(templateUrl)
-      .then((template) => {
-        const clause = new Clause(template);
-        clause.parse(template.getMetadata().getSample());
-        clause.draft({ wrapVariables: true })
-        .then((drafted) => {
-          const clauseMarkdown = `${markdownText}
-\`\`\` <clause src="${templateUrl}" clauseid="123">
-${drafted}
-\`\`\`
-This is some more text. Test moving a clause by dragging it or by using the up and down arrows. This is some more text. Test moving a clause by dragging it or by using the up and down arrows. This is some more text. Test moving a clause by dragging it or by using the up and down arrows. This is some more text. Test moving a clause by dragging it or by using the up and down arrows.
-`;
-          const slateValue = slateTransformer.fromMarkdown(clauseMarkdown);
-          setSlateValue(slateValue.document.children);
-        });
+      .then(async (template) => {
+        const sample = template.getMetadata().getSample();
+        const templatizedGrammar = template.getParserManager().getTemplatizedGrammar();
+        const modelManager = await template.getModelManager();
+        const data = templateMarkTransformer
+          .dataFromMarkdown(sample, templatizedGrammar, modelManager, 'contract');
+        const ciceroMark = templateMarkTransformer
+          .dataToCiceroMark(data, templatizedGrammar, modelManager, 'contract');
+        const slateValueNew = slateTransformer.fromCiceroMark(ciceroMark);
+
+        const extraMarkdown = `This is some more text. Test moving a clause by dragging it or by using the up and down arrows. This is some more text. Test moving a clause by dragging it or by using the up and down arrows. This is some more text. Test moving a clause by dragging it or by using the up and down arrows. This is some more text. Test moving a clause by dragging it or by using the up and down arrows.`;
+        const extraText = slateTransformer.fromMarkdown(extraMarkdown);
+        setSlateValue(slateValue.document.children);
+
+        const slateValueNewNew = {
+          document: {
+            data: {},
+            object: 'document',
+            children: [
+              ...slateValue,
+              {
+                children: slateValueNew.document.children,
+                data: {
+                  src: templateUrl,
+                  name: '123',
+                },
+                object: 'block',
+                type: 'clause',
+              },
+              ...extraText.document.children
+            ]
+          }
+        };
+        setSlateValue(slateValueNewNew.document.children);
```

Comment:
why do we need to construct the whole slate object in `slateValueNewNew` if we only need the children? Can we just construct the children array instead?

Reply:
Not sure what you mean. The `slateValueNew.document.children` is not wrapped in a clause, and we need `...slateValue` before the clause because the clause can't be the first item in the document's `children`

Reply:
But don't we only need to construct that array on line 70-82 and pass that array to `setSlateValue`?

Reply:
🤔 

You are correct, this is much better.